### DISCUSSION TITLE
How to input my Micro-CT images into Tutorial 01?

### DISCUSSION THREAD

**Discussion by ZZZYunpeng** [GeoChemFoam/GeoChemFoam#40] [Q&A] [answered]
Dear Professor,

Hi, professors and scholars. I hope you have a nice day.
I would like to consult some questions about tutorial 01.

I have a Micro-CT image, which is cylinder nor cube. I want to run some simulations based on tutorial 01. But when I modify the creatmesh.sh for creating my mesh, there are some errors in my simulation.
My simulation steps are as follows:
1. I confirm my image in Fiji, and make sure the width, height, number of images are correct.
![image](https://github.com/GeoChemFoam/GeoChemFoam/assets/140990976/7323cbad-619e-44cd-96fb-550c23110f31)

2. I modify the Brightness in Fiji so that make sure the pore_value=255, the solid_value=0.
![image](https://github.com/GeoChemFoam/GeoChemFoam/assets/140990976/06c5dfdf-0bba-4d69-826e-85fc5f7dffc3)

3. I copy and pack the image to $GCFOAM_IMG/raw. Then I modify the creatMesh.sh.
![image](https://github.com/GeoChemFoam/GeoChemFoam/assets/140990976/a16d755b-6141-4f72-9abd-2ca75c31736d)

4. After that, I run ./creatMesh.sh. But I met some errors.
![964e9dc8ec46c187526b474adfc420f](https://github.com/GeoChemFoam/GeoChemFoam/assets/140990976/8f6b5bc0-269c-4fae-a288-611c0703dbc4)

It seems that the wrong reason is the processing of stl files.
All my operations are based on https://github.com/GeoChemFoam/GeoChemFoam/wiki/Solving-permeability-on-an-open-access-binary-image
So I would like to know why drawing mesh is fail and how to change my settings.

Looking forward your reply. Thank you very much.

Best regards,
Yunpeng Zhang

**Comment by GeoChemFoam** [comment]
Have you opened it in paraview to see what could be going wrong?

On Jan 16, 2024, at 18:58, ZZZYunpeng ***@***.***> wrote:

﻿
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Dear Professor,

Hi, professors and scholars. I hope you have a nice day.
I would like to consult some questions about tutorial 01.

I have a Micro-CT image, which is cylinder nor cube. I want to run some simulations based on tutorial 01. But when I modify the creatmesh.sh for creating my mesh, there are some errors in my simulation.
My simulation steps are as follows:

  1.  I confirm my image in Fiji, and make sure the width, height, number of images are correct.
image.png (view on web)<https://github.com/GeoChemFoam/GeoChemFoam/assets/140990976/7323cbad-619e-44cd-96fb-550c23110f31>

  2.  I modify the Brightness in Fiji so that make sure the pore_value=255, the solid_value=0.
image.png (view on web)<https://github.com/GeoChemFoam/GeoChemFoam/assets/140990976/06c5dfdf-0bba-4d69-826e-85fc5f7dffc3>

  3.  I copy and pack the image to $GCFOAM_IMG/raw. Then I modify the creatMesh.sh.
image.png (view on web)<https://github.com/GeoChemFoam/GeoChemFoam/assets/140990976/a16d755b-6141-4f72-9abd-2ca75c31736d>

  4.  After that, I run ./creatMesh.sh. But I met some errors.
964e9dc8ec46c187526b474adfc420f.png (view on web)<https://github.com/GeoChemFoam/GeoChemFoam/assets/140990976/8f6b5bc0-269c-4fae-a288-611c0703dbc4>

It seems that the wrong reason is the processing of stl files.
All my operations are based on https://github.com/GeoChemFoam/GeoChemFoam/wiki/Solving-permeability-on-an-open-access-binary-image
So I would like to know why drawing mesh is fail and how to change my settings.

Looking forward your reply. Thank you very much.

Best regards,
Yunpeng Zhang

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/40>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFTXWCN2V73VACI2THDYO3EVHAVCNFSM6AAAAABB5JXCMWVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGA4DGNJRGE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Reply by ZZZYunpeng** [reply, marked as answer]
Dear GeoChemFoam Teams,

Thank you very much.
I have found the reason. Due to my computer's poor performance, it cannot calculate so many meshes. 
There are no errors in the code itself.

Best regards,
Yunpeng Zhang
